File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.java
import java.io.*;
import java.util.Arrays;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) throws IOException {
        Scanner scan = new Scanner(System.in);
        while(true){
            int[] arr = new int[3];
            int count = 0;
            for (int i = 0; i < 3; i++) {
                arr[i] = scan.nextInt();
                if(arr[i]==0) count++;
            }

            if(count == 3)break;
            // 입력 완료

            Arrays.sort(arr);
            if(arr[0]+arr[1]<=arr[2]){
                System.out.println("Invalid");
            }else{
                if(arr[0]==arr[1]&&arr[1]==arr[2]){
                    System.out.println("Equilateral");
                }else if(arr[0]==arr[1]||arr[1]==arr[2]||arr[0]==arr[2]){
                    System.out.println("Isosceles");
                }else{
                    System.out.println("Scalene");
                }
            }
        }
    }
}

File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.py
while True:
    array = []
    a,b,c = map(int,input().split())
    array.append(a)
    array.append(b)
    array.append(c)
    if a == b == c == 0:
        break;
    max_val = max(array)
    max_index = array.index(max_val)
    sum_val = 0
    for i in range(3):
        if i == max_index:
            pass
        else : sum_val+=array[i]
    if sum_val<=max_val:
        print("Invalid")
    elif len(set(array)) == 1:
        print("Equilateral")
    elif len(set(array)) == 2:
        print("Isosceles")
    else:
        print("Scalene")